ReactiveCocoa/Swift/SignalProducer.swift

```diff
@@ -549,6 +549,67 @@ public func concatMap<T, U, E>(transform: T -> SignalProducer<U, E>)(producer: S
     return producer |> map(transform) |> concat
 }
 
+/// Returns a signal that forwards values from the latest signal sent on
+/// `producer`, ignoring values sent on previous inner signals.
+///
+/// An error sent on `producer` or the latest inner signal will be sent on the
+/// returned signal.
+///
+/// The returned signal completes when `producer` and the latest inner signal
+/// have both completed.
+public func latest<T, E>(producer: SignalProducer<SignalProducer<T, E>, E>) -> SignalProducer<T, E> {
+	return SignalProducer<T, E> { sink, disposable in
+		let producerCompleted = Atomic(false)
+		let latestCompleted = Atomic(false)
+		
+		let completeIfNecessary: () -> () = {
+			if producerCompleted.value && latestCompleted.value {
+				sendCompleted(sink)
+			}
+		}
+
+		let latestDisposable = SerialDisposable()
+		disposable.addDisposable(latestDisposable)
+
+		producer.startWithSignal { signal, producerDisposable in
+			disposable.addDisposable(producerDisposable)
+			
+			signal.observe(
+				next: { signal in
```

Comment:
Shadowing `signal` here and below makes this code a bit harder to reason over. Especially since this one is a `SignalProducer` and not a `Signal`. Maybe `innerProducer` or `latestProducer` here and similarly with its produced signal below